### PR TITLE
feat(feature-flag): add feature flag for Circle

### PR DESCRIPTION
**Pull request by devformatters**

### CODE REVIEW

File: src/views/ArticleDetail/EditMode/index.tsx [hunk 1]
```diff
@@ -166,7 +168,7 @@ const EditMode: React.FC<EditModeProps> = ({ article, onCancel, onSaved }) => {
             disabled={isEditDisabled}
           />
 
-          {toggleCircle && (
+          {toggleCircle && features.circle_management && (
```
Comment (robertu7) [inline src/views/ArticleDetail/EditMode/index.tsx:171]: Forgot adding  `features.circle_management` at `<BottomBar>` L236?

File: src/components/Hook/useFeatures.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+/**
+ * This hook is a wrapper of accessing feature flag context.
+ *
+ * Usage:
+ *
+ * ```jsx
+ *   const features = useFeatures()
+ * ```
+ */
+export const useFeatures = () => {
```
Comment (robertu7) [inline src/components/Hook/useFeatures.ts:14]: Nice abstraction, would it be good if applying to the rest of the places, don't leave legacy :)

**Review by robertu7 — CHANGES_REQUESTED** [review]

File: src/views/ArticleDetail/EditMode/index.tsx (hunk 1 above)
Comment (devformatters) [reply src/views/ArticleDetail/EditMode/index.tsx:171]: Totally missed it. Now add it.

File: src/components/Hook/useFeatures.ts (hunk 2 above)
Comment (devformatters) [reply src/components/Hook/useFeatures.ts:14]: Done.

**Review by robertu7 — APPROVED** [review]

**Pull request merged**